src/kubernetes.rs

```diff
@@ -1,27 +1,65 @@
 use {
-    crate::k8s_helpers::{self, SecretType},
-    k8s_openapi::api::core::v1::{Namespace, Secret},
+    crate::{
+        docker::DockerImage,
+        k8s_helpers::{self, SecretType},
+        validator_config::ValidatorConfig,
+        ValidatorType,
+    },
+    k8s_openapi::{
+        api::{
+            apps::v1::ReplicaSet,
+            core::v1::{
+                EnvVar, EnvVarSource, Namespace, ObjectFieldSelector, Secret, SecretVolumeSource,
+                Volume, VolumeMount,
+            },
+        },
+        apimachinery::pkg::api::resource::Quantity,
+    },
     kube::{
         api::{Api, ListParams, PostParams},
         Client,
     },
-    std::{
-        collections::{BTreeMap, HashMap},
-        error::Error,
-        path::Path,
-    },
+    log::*,
+    solana_sdk::{pubkey::Pubkey, signature::keypair::read_keypair_file, signer::Signer},
+    std::{collections::BTreeMap, error::Error, path::Path},
 };
 
-pub struct Kubernetes {
+#[derive(Debug, Clone)]
+pub struct PodRequests {
+    requests: BTreeMap<String, Quantity>,
+}
+
+impl PodRequests {
+    pub fn new(cpu_requests: String, memory_requests: String) -> PodRequests {
+        PodRequests {
+            requests: vec![
+                ("cpu".to_string(), Quantity(cpu_requests)),
+                ("memory".to_string(), Quantity(memory_requests)),
+            ]
+            .into_iter()
+            .collect(),
```

Comment:
nit: you should be able to use `BTreeMap::from`, ie
```suggestion
            requests: BTreeMap::from([
                ("cpu".to_string(), Quantity(cpu_requests)),
                ("memory".to_string(), Quantity(memory_requests)),
            ])
```

https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#impl-From%3C%5B(K,+V);+N%5D%3E-for-BTreeMap%3CK,+V%3E